src/resolve/resolved-dns-synthesize.c

```diff
@@ -444,6 +445,11 @@ int dns_synthesize_answer(
                 } else if (dns_name_address(name, &af, &address) > 0) {
                         int v, w;
 
+                        if (getenv_bool("SYSTEMD_RESOLVED_SYNTHESIZE_HOSTNAME") == 0) {
+                                nxdomain = true;
```

Comment:
hm, nxdomain seems a bit drastic? i.e. if you return nxdomain then this ends the lookup logic. But it appears to me, if the env var is set we simply shouldn't respond at a ll, i.e. just do `continue` here...

also, shouldn't the manager_is_own_hostname() branch above get the same treatment? to disable both directions of local hostname resolution? from address to name and from name to address?

Reply:
For the `nxdomain`, you are right. That is too drastic. I will just `continue` iteration as suggested.

I guess both directions makes sense. I will do it for normal and reverse lookups (and change ENVIRONMENT.md).